dids/diddht/diddht.go

```diff
@@ -247,9 +247,9 @@ func CreateWithContext(ctx context.Context, opts ...CreateOption) (did.BearerDID
 		document.AddVerificationMethod(newVM, didcore.Purposes(pk.purposes...))
 	}
 
-	for _, service := range o.services {
-		s := service
-		document.AddService(&s)
+	for i := range o.services {
```

Comment:
why i changed this:

this block:
```go
for _, service := range o.services {
		s := service
		document.AddService(&s)
	}
```
results in error  `The copy of the 'for' variable "service" can be deleted (Go 1.22+)`

so if i change it to this:
```go
for _, service := range o.services {	
		document.AddService(&service)
	}
```
results in a different error `G601: Implicit memory aliasing in for loop`

so the best way to make linter happy, not introduce bug (the second one will result in all pointers pointing to the last item in the slice by the time the loop exits), and make code readable, is to use index to iterate but set a variable as to not write something awful like `document.AddService(&o.services[i])`

Reply:
```
G601: Implicit memory aliasing in for loop
```

This is no longer valid and should be suppressed in `.golangci.yml` - your first attempt is correct.